lib/ecto/changeset.ex

```diff
@@ -2067,6 +2065,42 @@ defmodule Ecto.Changeset do
     end
   end
 
+  @doc """
+  Returns a list of fields that are present.
+
+  Each field passed into this function will have its prescence evaluated
+  according to the same rules as `validate_required/3`. If it is present,
+  it is returned. If it is not present, it is removed from the list.
+
+  This is useful when performing complex validations that are not possible with
+  `validate_required/3`. For example, evaluating whether at least one field
+  from a list is present or evaluating that exactly one field from a list is
+  present.
+
+  ## Examples
+
+      iex> changeset = cast(%Post{}, %{title: "Title"}, [:title])
+      iex> filter_required(changeset, [:title, :body])
+      [:title]
+
+      iex> changeset = cast(%Post{}, %{color: "Red"}, [:color])
+      iex> filter_required(changeset, [:title, :body])
+      []
+
+  """
+  @spec filter_required(t, list | atom) :: list
+  def filter_required(%Changeset{} = changeset, fields) when not is_nil(fields) do
+    %{types: types} = changeset
+    fields = List.wrap(fields)
+
+    Enum.filter(fields, &(not invalid_required_field!(changeset, types, &1)))
+  end
+
+  defp invalid_required_field!(changeset, types, field) do
+    ensure_field_not_many!(types, field) && missing?(changeset, field) &&
```

Comment:
`ensure_field_not_many!` warns with a message mentioning calling `validate_required` which would not be correct when the user called filter_required. Could you look into that?

Reply:
Thanks good catch. How does the change I made look?